modules/data.atmosphere/R/download.MACA.R

```diff
@@ -0,0 +1,145 @@
+  lat_MACA = round(lat*24)
+  lon.in = as.numeric(lon.in)
+  if (lon.in < 0){
+    lon.in = 180 - lon.in}
+  lon = lon.in - 235.22784423828125
+  lon_MACA = round(lon*24)
+  
+
+  dap_base ='http://thredds.northwestknowledge.net:8080/thredds/dodsC/MACAV2'
+
+  ylist <- seq(start_year,end_year,by=1)
+  rows = length(ylist)
+  results <- data.frame(file=character(rows), host=character(rows),
+                        mimetype=character(rows), formatname=character(rows),
+                        startdate=character(rows), enddate=character(rows),
+                        dbfile.name = paste("MACA",model,scenario,ensemble_member,sep="."),#"MACA",
+                        stringsAsFactors = FALSE)
+  
+  
+  prevar = data.frame(DAP.name = c("tasmax","tasmin","rsds","uas","vas","huss","pr"), 
+                      CF.name = c("air_temperature","air_temperature","surface_downwelling_shortwave_flux_in_air","eastward_wind","northward_wind","specific_humidity","precipitation"),
+                      units = c('Kelvin','Kelvin',"W/m2","m/s","m/s","g/g","kg/m2/s")
+                      )
+  var = data.frame(DAP.name = c("air_temperature", "rsds","uas","vas","huss","pr"),
+                   CF.name = c("air_temperature","surface_downwelling_shortwave_flux_in_air","eastward_wind","northward_wind","specific_humidity","precipitation_flux"),
+                   units = c('Kelvin',"W/m2","m/s","m/s","g/g","kg/m2/s")
+  )
+  
+  for (i in 1:rows){
+    year = ylist[i]    
+    ntime = (1825)
+    
+    met_start = 2006
+    met_block = 5
+    url_year = met_start + floor((year-met_start)/met_block)*met_block
+    start_url = paste0(url_year)
+    end_url = paste0(url_year+met_block-1)
+    dir.create(outfolder, showWarnings=FALSE, recursive=TRUE)
+    
+    loc.file = file.path(outfolder,paste("MACA",model,scenario,ensemble_member,year,"nc",sep="."))
+    
+    ## Create dimensions
+    lat <- ncdim_def(name='latitude', units='degree_north', vals=lat.in, create_dimvar=TRUE)
+    lon <- ncdim_def(name='longitude', units='degree_east', vals=lon.in, create_dimvar=TRUE)
+    time <- ncdim_def(name='time', units="sec", vals=(1:365)*86400, create_dimvar=TRUE, unlim=TRUE)
+    dim=list(lat,lon,time)
+    
+    prevar.list = list()
+    predat.list = list()
+    
+    ## get data off OpenDAP
+    for(j in 1:length(prevar$DAP.name)){
+      dap_end = paste0('/',model,'/macav2metdata_',prevar$DAP.name[j],'_',model,'_',ensemble_member,'_',scenario,'_',start_url,'_',end_url,'_CONUS_daily.nc')
+      dap_file = paste0(dap_base,dap_end)
+      dap = nc_open(dap_file)
+      predat.list[[j]] = ncvar_get(dap,as.character(prevar$CF.name[j]),c(lon_MACA,lat_MACA,1),c(1,1,ntime))
+      prevar.list[[j]] = ncvar_def(name=as.character(prevar$CF.name[j]), units=as.character(prevar$units[j]), dim=dim, missval=-9999.0, verbose=verbose)
+      nc_close(dap)
+    }
+    
+    #take average of temperature min and max
+    predat.list[[1]] = (predat.list[[1]]+predat.list[[2]])/2
```

Comment:
@mdietze Yeah, it has Tmin and Tmax, but I wasn't sure how to include them because there is no CF convention for them. But now that I'm working on temporal downscaling, I might force MACA to undergo it just so we can have the Tmin and Tmax in the downscaled product.